src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,131 @@
+from autotester.server.utils.file_management import (
+    clean_dir_name,
+    random_tmpfile_name,
+    recursive_iglob,
+    copy_tree,
+    copy_test_script_files,
+    ignore_missing_dir_error,
+    move_tree,
+    fd_open,
+    fd_lock,
+)
+from autotester.server.utils import redis_management
+import os.path
+import re
+import tempfile
+import shutil
+from autotester.config import config
+import subprocess
+
+FILES_DIRNAME = config["_workspace_contents", "_files_dir"]
+
+
+def test_clean_dir_name():
+    a = "markus/address"
+    b = a.replace("/", "_")
+    assert clean_dir_name(a) == b
+
+
+def test_random_tmpfile_name():
+    tmp_file = random_tmpfile_name()
+    dir_name = os.path.dirname(tmp_file)
+    base_name = os.path.basename(tmp_file)
+    assert os.path.exists(dir_name) and re.match(
+        r"[+\-]" "0" "[xX]" "0|" "([1-9A-Fa-f][1-9A-Fa-f]*)", base_name
+    )
+
+
+def test_recursive_iglob():
```

Comment:
This looks good overall. But it's only testing a single directory structure. In general it's good to write tests that check that the function works as intended for multiple possible structures.

Some ideas (that you can feel free to add to):

- when the directory is empty
- when the directory contains a single file
- when the directory contains a single directory
- when the directory contains both files and directories (this is your current test)
- when files are nested in subdirectories more than 2 directories deep
- ...